controllers/vaultdynamicsecret_controller.go

```diff
@@ -155,14 +151,17 @@ func (r *VaultDynamicSecretReconciler) Reconcile(ctx context.Context, req ctrl.R
 				// compatible with computeHorizonWithJitter()
 				leaseDuration = time.Second * 5
 			}
-			horizon := computeHorizonWithJitter(leaseDuration)
+			horizon := computeDynamicHorizonWithJitter(leaseDuration, o.Spec.RenewalPercent)
 			r.Recorder.Eventf(o, corev1.EventTypeNormal, consts.ReasonSecretLeaseRenewal,
 				"Renewed lease, lease_id=%s, horizon=%s", leaseID, horizon)
 			return ctrl.Result{RequeueAfter: horizon}, nil
 		} else {
 			// The secretLease was not renewed or failed, continue through Reconcile and do a rollout restart.
 			doRolloutRestart = true
-			if !isLeaseNotfoundError(err) {
+			if e, ok := err.(*LeaseTruncatedError); ok {
```

Comment:
We may want to handle the case were err is wrapped:

```suggestion
			if e, ok := err.(*LeaseTruncatedError); ok || e != nil && errors.As(err, &e)
```

Reply:
Added in 73610bc.